itest.py

```diff
@@ -0,0 +1,832 @@
+    # match/case candidate?
+    if test_args is None:
+        if desired_test.stem == 'run_plugin_tests':
+            # Gather valid plugins
+            yes_plugins = tuple(get_supported_plugins())
+            assert(len(yes_plugins) > 0)
+
+            for plugin in yes_plugins:
+                print(plugin)
+    
+            # Prompt user for desired plugin
+            plugin_choice = input("What plugin ya wanna build? ")
+            validate_choice(plugin_choice, yes_plugins)
+
+            # Add plugin git name
+            bonus_args.append(plugin_choice)
+
+            # Dumb trim of plugin
+            for prefix in PLUGIN_PREFIXES:
+                if plugin_choice.startswith(prefix):
+                    plugin_name = plugin_choice[len(prefix)+1:]
+
+            # Append plugin dir
+            named_args.append(('--plugin-package-directory', str(IRODS_PLUGIN_PACKAGE_DIR).format(os_name=os_choice, short_plugin_name=plugin_name)))
+        elif desired_test.stem == 'run_core_tests':
+            num_executors = int(input('How many test executors do you want? '))
+            named_args.append(('--concurrent-test-executor-count', str(num_executors)))
+        elif desired_test.stem == 'run_topology_tests':
+            num_executors = int(input('How many test executors do you want? '))
+            named_args.append(('--concurrent-test-executor-count', str(num_executors)))
+
+            topo_options = ['provider', 'consumer']
+            for option in topo_options:
+                print(option)
+
+            choice = input('Where ya wanna run the tests? ')
+            validate_choice(choice, topo_options)
+
+            bonus_args.append(choice)
+
+    else:
+        test_idx = len(test_args) 
+        try:
+            test_idx = test_args.index('--tests')
+        except ValueError:
+            ...
+        bonus_args = test_args[:test_idx]
+
+    arg_list = [sys.executable, str(desired_test),]
+    for arg, param in named_args:
+        arg_list.append(arg)
+
+        if isinstance(param, PurePath):
+            param = str(param).format(os_name=os_choice)
+
+        arg_list.append(param)
+
+    arg_list.extend(bonus_args)
+
+    failed_tests_list = None
+    if test_args is None:
+        tests_to_run = input('Please input test you wish to run (default: all): ').split()
+
+        if len(tests_to_run) != 0:
+            failed_tests_list = tests_to_run
+            failed_tests_list.insert(0, '--tests')
+    else:
+        failed_tests_list = test_args[test_idx:]
+
+    res = None
+    failed_tests_pattern = re.compile('(?<=List of failed tests:\n\t).*')
+
+    max_duplicate_attempts = 3
+    current_duplicate_attempt = 1
+    attempt = 0
+
+    while current_duplicate_attempt <= max_duplicate_attempts:
```

Comment:
Consider modifying the while loop to cap the attempts to 1. People can opt-in to the auto-retest.

Maybe can skip this change and have it be opt-out?

Reply:
Ignoring for now.